sale_loyalty_limit/models/sale_order.py

```diff
@@ -0,0 +1,132 @@
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _program_check_compute_points(self, programs):
+        res = super()._program_check_compute_points(programs)
+        programs_with_additional_errors = {}
+        # Iterate through the programs that initially have no errors
+        for program, result in res.items():
+            if "error" not in result:
```

Comment:
Avoid nesting if not necessary
```suggestion
            if result == "error":
                continue
```